source/fuzz/transformation_replace_irrelevant_id.h

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef SOURCE_FUZZ_TRANSFORMATION_REPLACE_IRRELEVANT_ID_H_
+#define SOURCE_FUZZ_TRANSFORMATION_REPLACE_IRRELEVANT_ID_H_
+
+#include "source/fuzz/transformation.h"
+
+namespace spvtools {
+namespace fuzz {
+
+class TransformationReplaceIrrelevantId : public Transformation {
+ public:
+  explicit TransformationReplaceIrrelevantId(
+      const protobufs::TransformationReplaceIrrelevantId& message);
+
+  TransformationReplaceIrrelevantId(
+      const protobufs::IdUseDescriptor& id_use_descriptor,
+      uint32_t replacement_id);
+
+  // - The id of interest in |message_.id_use_descriptor| is irrelevant
+  //   according to the fact manager.
+  // - The types of the original id and of the replacement ids are the same.
+  // - |message_.replacement_id| is available to use at the enclosing
+  //   instruction of |message_.id_use_descriptor|.
+  // - The original id is in principle replaceable with any other id of the same
+  //   type. In particular:
+  //    - it is not used to index a struct in an access chain (such ids must be
+  //      constants).
+  //    - it is not used as the Sample parameter of an OpImageTexelPointer
+  //      instruction (this parameter must satisfy particular requirements).
```

Comment:
Perhaps this comment and the corresponding comment in transformation_replace_id_with_synonym should be shortened so that, instead of "In particular:" onwards they say "See ... for details.", where "..." is the name of the helper function.

Reply:
The motivation is to reduce comment maintenance burden / the chance of comments becoming out of sync or stale.

Reply:
Done, both here and in transformation_replace_id_with_synonym